spec/requests/api/v2/media_spec.rb

```diff
@@ -2,17 +2,99 @@
 
 require 'rails_helper'
 
-RSpec.describe 'Media API', paperclip_processing: true do
+RSpec.describe 'Media', paperclip_processing: true do
   let(:user)    { Fabricate(:user) }
   let(:token)   { Fabricate(:accessible_access_token, resource_owner_id: user.id, scopes: scopes) }
-  let(:scopes)  { 'write' }
+  let(:scopes)  { 'write:media' }
   let(:headers) { { 'Authorization' => "Bearer #{token.token}" } }
 
   describe 'POST /api/v2/media' do
-    it 'returns http success' do
-      post '/api/v2/media', headers: headers, params: { file: fixture_file_upload('attachment-jpg.123456_abcd', 'image/jpeg') }
-      expect(File.exist?(user.account.media_attachments.first.file.path(:small))).to be true
-      expect(response).to have_http_status(200)
+    subject do
+      post '/api/v2/media', headers: headers, params: params
+    end
+
+    let(:params) { {} }
+
+    shared_examples 'a successful media upload' do |media_type|
+      it 'uploads the file successfully' do
+        subject
+
+        aggregate_failures do
+          expect(response).to have_http_status(200)
+          expect(MediaAttachment.first).to be_present
+          expect(MediaAttachment.first).to have_attached_file(:file)
+        end
+      end
+
+      it 'returns the correct media content' do
+        subject
+
+        body = body_as_json
+
+        expect(body).to match(
+          a_hash_including(id: MediaAttachment.first.id.to_s, description: params[:description], type: media_type)
+        )
+      end
+    end
+
+    it_behaves_like 'forbidden for wrong scope', 'read read:media'
+
+    describe 'when paperclip errors occur' do
+      let(:media_attachments) { double }
+      let(:params)            { { file: fixture_file_upload('attachment.jpg', 'image/jpeg') } }
+
+      before do
+        allow(User).to receive(:find).with(token.resource_owner_id).and_return(user)
```

Comment:
Is that necessary? Isn't the token already correctly defined?

Reply:
That's necessary in order to stub `user.account.media_attachments` so that it returns the `media_attachments` double.
I'm not quite sure if that's the best way to simulate those scenarios or if it's the best way to stub these methods, though.